onnx/backend/test/case/node/__init__.py

```diff
@@ -103,27 +103,21 @@ def function_testcase_helper(node, input_types, name):  # type: (NodeProto, List
     return node_list
 
 
-def _extract_value_info(input, name, ele_type=None):  # type: (Union[List[Any], np.ndarray, None], Text, np.dtype) -> onnx.ValueInfoProto
-    if isinstance(input, list):
-        # TODO: Account for recursive sequence case. Right now, this function supports
-        # Sequences of Tensors.
-        return onnx.helper.make_tensor_sequence_value_info(
-            name=name,
-            elem_type=ele_type if ele_type else onnx.mapping.NP_TYPE_TO_TENSOR_TYPE[input[0].dtype],
-            shape=None
-        )
-    elif input is None:
-        if ele_type is None:
-            raise NotImplementedError("extract_value for optional tensor with no data type not "
-                                      "supported.")
+def _extract_value_info(input, name, type_proto=None):  # type: (Union[List[Any], np.ndarray, None], Text, Optional[TypeProto]) -> onnx.ValueInfoProto
+    if type_proto is None:
+        if input is None:
+            raise NotImplementedError("_extract_value_info: both input and type_proto arguments cannot be None.")
+        elif isinstance(input, list):
+            elem_type = onnx.mapping.NP_TYPE_TO_TENSOR_TYPE[input[0].dtype]
+            shape = input[0].shape
```

Comment:
shape should be None as before. We can't assume all elements in the sequence have the same shape (unless we check and verify this).